packages/react/src/ReactCreateEventComponent.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ * @flow
+ */
+
+import type {ReactEventComponent, ReactEventResponder} from 'shared/ReactTypes';
+import {enableEventAPI} from 'shared/ReactFeatureFlags';
+
+import {REACT_EVENT_COMPONENT_TYPE} from 'shared/ReactSymbols';
+
+type EventComponentConfig = {
+  displayName: string,
+  props: null | Object,
+  responder: ReactEventResponder,
+};
+
+export function createEventComponent({
+  displayName,
```

Comment:
I think it was maybe better how it was before, as we're creating less objects:

```
export function createEventComponent(responder: ReactEventResponder, displayName?: string)
```

We should also freeze the `responder` too:

```
if (__DEV__) {
  Object.freeze(responder);
}
```

Reply:
Sorry, I forgot it got moved: https://github.com/facebook/react/blob/master/packages/react-dom/src/events/DOMEventResponderSystem.js#L466

Reply:
> I don't think using an object for a one-off function call is going to be a problem. We used an object for export previously. But happy to change it.

It's not a big deal, but just means less bytes and one less object created which seems better given the API for this function only has 2 arguments now